### PR TITLE
Do not strip tab characters from data_column reference and clarify documentation for use_header_names

### PR DESCRIPTION
**Pull request by bernt-matthias**
- add a test (also as example)
- clarify in the docs
- bugfix empty columns must not be stripped

### CODE REVIEW

File: lib/galaxy/tools/parameters/basic.py [hunk 1]
```diff
@@ -1271,7 +1271,7 @@ def get_options(self, trans, other_values):
             try:
                 with open(dataset.get_file_name(), 'r') as f:
                     head = f.readline()
-                cnames = head.rstrip().split('\t')
+                cnames = head.rstrip("\n\r ").split('\t')
```
Comment (bgruening) [inline lib/galaxy/tools/parameters/basic.py:1274]: I would assume this as the default. The documentation says "whitespaces", is \r not included?

Comment (bernt-matthias) [reply lib/galaxy/tools/parameters/basic.py:1274]: Default is that `\n\r\t `. The problem is that this removes the first or last columns if they are empty.

File: test/functional/tools/column_param.xml [hunk 2]
```diff
@@ -1,10 +1,13 @@
 <tool id="column_param" name="Column Param">
-  <command>
-    cut -f '$col' '$input1' > '$output1'
-  </command>
+  <command><![CDATA[
+    echo "col $col" >> '$output1' &&
```
Comment (jmchilton) [inline test/functional/tools/column_param.xml]: This is odd to add to that file right - can you echo these to a new second output?

Comment (bernt-matthias) [reply test/functional/tools/column_param.xml]: Good idea.

**Review by jmchilton — APPROVED** [review]

**Comment by jmchilton** [comment]
Looks awesome - can you merge this once the framework tests pass? Thanks so much for the enhancements.

**Comment by bernt-matthias** [comment]
Hmm. framework tests timed out :(

File: test/functional/tools/column_param.xml [hunk 3]
```diff
@@ -1,23 +1,34 @@
 <tool id="column_param" name="Column Param">
-  <command>
-    cut -f '$col' '$input1' > '$output1'
-  </command>
+  <command><![CDATA[
+    cut -f '$col' '$input1' >> '$output1'
+    echo "col $col" >> '$output2' &&
+    echo "col_names $col_names" >> '$output2' &&
```
Comment (nsoranzo) [inline test/functional/tools/column_param.xml]:
```suggestion
cut -f '$col' '$input1' > '$output1' &&
echo "col $col" > '$output2' &&
echo "col_names $col_names" >> '$output2'
```

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 4]
```diff
@@ -2563,8 +2563,10 @@ Set to ``false`` to not force user to select an option in the list.</xs:document
             <xs:documentation xml:lang="en">Used only if the ``type`` attribute
 value is ``data_column``, if ``true`` Galaxy assumes first row of ``data_ref``
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/xsd/galaxy.xsd]:
```suggestion
value is ``data_column``. If ``true``, Galaxy assumes the first row of ``data_ref``
```

File: test/functional/tools/column_param.xml [hunk 5]
```diff
@@ -1,23 +1,34 @@
...
+echo "col_names $col_names" >> '$output2'
+  ]]></command>
   <inputs>
     <param type="data" format="tabular" name="input1" label="Input 1" />
     <param name="col" type="data_column" data_ref="input1" label="Column to Use" />
+    <param name="col_names" type="data_column" data_ref="input1" use_header_names="true" label="Column to Use" />
   </inputs>
   <outputs>
     <data name="output1" format="tabular" />
+    <data name="output2" format="tabular" />
```
Comment (nsoranzo) [inline test/functional/tools/column_param.xml]:
```suggestion
    <data name="output2" format="txt" />
```

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**